express-notes-api/index.js

```diff
@@ -0,0 +1,103 @@
+  const temp = [];
+  for (const x in json.notes) {
+    temp.push(json.notes[x]);
+  }
+  res.status(200).send(temp);
+});
+
+app.get('/api/notes/:id', function (req, res) {
+  const id = req.params.id;
+  const num = Number(req.params.id);
+  if (Number.isInteger(num) && num > 0) {
+    if (!json.notes[id]) {
+      const error = { error: `Cannot find Id of ${req.params.id}` };
+      res.status(404).send(error);
+    } else {
+      res.status(200).send(json.notes[id]);
+    }
+  } else if (req.params.id <= 0) {
+    const error = { error: 'Input is not a positive integer.' };
+    res.status(400).send(error);
+  }
+});
+
+app.post('/api/notes', function (req, res) {
+  if (Object.keys(req.body).length === 0) {
+    const error = { error: 'Content is a required field' };
+    res.status(400).send(error);
+  } else {
+    const newNote = req.body;
+    const id = json.nextId;
+    newNote.id = id;
+    json.notes[json.nextId] = newNote;
+    json.nextId++;
+    fs.writeFile('./data.json', JSON.stringify(json, null, 2), err => {
+      const error = { error: 'An unexpected error occurred.' };
+      if (err) console.error(error);
+    });
+    res.status(201).json(newNote);
+  }
+});
+
+app.delete('/api/notes/:id', function (req, res) {
+  const id = req.params.id;
+  const num = Number(req.params.id);
+  if (Number.isInteger(num) && num > 0) {
+    if (!json.notes[id]) {
+      const error = { error: `Cannot find Id of ${req.params.id}` };
+      res.status(404).send(error);
+    } else {
+      delete json.notes[id];
+      fs.writeFile('./data.json', JSON.stringify(json, null, 2), err => {
+        const error = { error: 'An unexpected error occurred.' };
+        if (err) console.error(error);
+      });
+      res.status(204).send();
+    }
+  } else if (req.params.id <= 0) {
+    const error = { error: 'Input is not a positive integer.' };
+    res.status(400).send(error);
+  }
+});
+
+app.put('/api/notes/:id', function (req, res) {
+  const id = req.params.id;
+  const num = Number(req.params.id);
+  if (Number.isInteger(num) && num > 0 && Object.keys(req.body).length !== 0) {
+    if (!json.notes[id]) {
+      const error = { error: `Cannot find Id of ${req.params.id}` };
+      res.status(404).send(error);
+    } else {
+      const newNote = req.body;
+      const id = json.notes[req.params.id].id;
+      newNote.id = id;
+      json.notes[id] = newNote;
+      fs.writeFile('./data.json', JSON.stringify(json, null, 2), err => {
+        if (err) {
+          const error = { error: 'An unexpected error occurred.' };
+          res.status(500).send(error);
+        }
+      });
+      res.status(200).send(json.notes[id]);
+    }
+  } else if (req.params.id <= 0) {
+    const error = { error: 'Input is not a positive integer.' };
+    res.status(400).send(error);
+  } else if (Object.keys(req.body).length === 0) {
+    const error = { error: 'Content is a required field' };
+    res.status(400).send(error);
+  }
```

Comment:
Just an else here would be better, with the else if you aren't catching everything and you have already checked for this stuff. You can use an error message that covers everything. `Id must be a positive integer and content is a required field`